SHSearchBar/Classes/SHSearchBar.swift

```diff
@@ -332,11 +335,13 @@ extension SHSearchBar: UITextFieldDelegate {
     }
 
     public func textFieldShouldClear(_ textField: UITextField) -> Bool {
+        clearButtonDidClick?()
         let shouldClear = delegate?.searchBarShouldClear(self) ?? searchBarShouldClear(self)
         return shouldClear
     }
 
     public func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        clearButtonDidClick?()
         let shouldReturn = delegate?.searchBarShouldReturn(self) ?? searchBarShouldReturn(self)
```

Comment:
Just below this line the delegate function is called to inform the outside world that the textfield returned. Please use this function since it is tested.